azure-pipelines.yml

```diff
@@ -49,16 +40,15 @@ jobs:
       hypothesis
       matplotlib
       numpy
-      pytest
+      pytest==6.1.2
       pytest-cov
       pytest-xdist
       scikit-learn
       scipy
-      h5py==2.10.0
+      h5py
```

Comment:
Are we  using azure _only_ for 32-bit Windows builds now? If so we could pin h5py==2.10.0 here (+ comment) although I don't understand why the package resolving algorithms are not able to get the right version by itself. Is this a pip or conda issue? (Naively, I thought that if h5py is not available for an arch then it will just use 2.10.0.)

Reply:
> Are we using azure only for 32-bit Windows builds now?

I don't think so--it looks like PRs to `develop` have more matrix entries than that for azure. It looks like this PR is deleting some Azure matrix entries--can't tell if that's because of the changes to github workflows CI here though.

I think the `pip` resolver is looking for the latest available version (it is definitely pip, not conda here), and it will build from source if it has to (and it fails to do that here). One option might be to use `--only-binary` flag for `pip install` for `h5py`; that should exclude usage of source packages and fall back to newest available wheel: https://pip.pypa.io/en/stable/reference/pip_install/#install-only-binary (assuming the older h5py has the 32-bit binary and wasn't just building that from source..)

I'm probably on the other side for 32-bit support--if deps don't support it, shed the deps/related feature or don't take it on in the first place or at least make it fully optional, especially for Windows where 32-bit is still important.